build_projects/shared-build-targets-utils/Publishing/AzurePublisher.cs

```diff
@@ -183,12 +183,37 @@ public string CalculateArchiveUploadUrl(string archiveFile, string channel, stri
             return $"{s_dotnetBlobRootUrl}{CalculateArchiveBlob(archiveFile, channel, version)}";
         }
 
-        public string CalculateArchiveBlob(string archiveFile, string channel, string version)
+        public static string CalculateArchiveBlob(string archiveFile, string channel, string version)
         {
             return $"{channel}/Binaries/{version}/{Path.GetFileName(archiveFile)}";
         }
 
-        public void DownloadFiles(string blobVirtualDirectory, string fileExtension, string downloadPath)
+        public static void DownloadFile(string blobFilePath, string localDownloadPath)
+        {
+            var blobUrl = $"{s_dotnetBlobRootUrl}{blobFilePath}";
+
+            using (var client = new HttpClient())
+            {
+                client.Timeout = TimeSpan.FromMinutes(1);
```

Comment:
5 would be better than 1.  How big is the shared framework zip file?  40 MB?  Why do we care about hanging now?  We don't care about a timeout when downloading Stage0, do we?

---
In reply to: [64613316](https://github.com/dotnet/cli/pull/3230#discussion_r64613316) [](ancestors = 64613316)